packages/components/src/mobile/keyboard-aware-flat-list/index.ios.js

```diff
@@ -19,53 +15,62 @@ export const KeyboardAwareFlatList = ( {
 	autoScroll,
 	scrollViewStyle,
 	inputAccessoryViewHeight,
+	scrollEnabled,
 	...listProps
-} ) => (
-	<KeyboardAwareScrollView
-		style={ [ { flex: 1 }, scrollViewStyle ] }
-		keyboardDismissMode="none"
-		enableResetScrollToCoords={ false }
-		keyboardShouldPersistTaps="handled"
-		extraScrollHeight={ extraScrollHeight }
-		extraHeight={ 0 }
-		inputAccessoryViewHeight={ inputAccessoryViewHeight }
-		enableAutomaticScroll={ autoScroll === undefined ? false : autoScroll }
-		ref={ ( ref ) => {
-			this.scrollViewRef = ref;
-			innerRef( ref );
-		} }
-		onKeyboardWillHide={ () => {
-			this.keyboardWillShowIndicator = false;
-		} }
-		onKeyboardDidHide={ () => {
-			setTimeout( () => {
-				if (
-					! this.keyboardWillShowIndicator &&
-					this.latestContentOffsetY !== undefined &&
-					! shouldPreventAutomaticScroll()
-				) {
-					// Reset the content position if keyboard is still closed
-					if ( this.scrollViewRef ) {
-						this.scrollViewRef.scrollToPosition(
-							0,
-							this.latestContentOffsetY,
-							true
-						);
+} ) => {
+	const scrollViewRef = useRef();
+	const keyboardWillShowIndicator = useRef();
+	const latestContentOffsetY = useRef();
+	return (
+		<RNKeyboardAwareFlatList
+			style={ [ { flex: 1 }, scrollViewStyle ] }
+			contentContainerStyle={ { flexGrow: 1, backgroundColor: 'red' } }
+			keyboardDismissMode="none"
+			enableResetScrollToCoords={ false }
+			keyboardShouldPersistTaps="handled"
+			extraScrollHeight={ extraScrollHeight }
+			extraHeight={ 0 }
+			inputAccessoryViewHeight={ inputAccessoryViewHeight }
+			enableAutomaticScroll={
+				autoScroll === undefined ? false : autoScroll
+			}
+			ref={ ( ref ) => {
+				scrollViewRef.current = ref;
+				innerRef( ref );
+			} }
+			onKeyboardWillHide={ () => {
+				keyboardWillShowIndicator.current = false;
+			} }
+			onKeyboardDidHide={ () => {
+				setTimeout( () => {
+					if (
+						! keyboardWillShowIndicator.current &&
+						latestContentOffsetY.current !== undefined &&
+						! shouldPreventAutomaticScroll()
+					) {
+						// Reset the content position if keyboard is still closed
+						if ( scrollViewRef.current ) {
+							scrollViewRef.current.scrollToPosition(
+								0,
+								latestContentOffsetY.current,
+								true
+							);
+						}
 					}
-				}
-			}, 50 );
-		} }
-		onKeyboardWillShow={ () => {
-			this.keyboardWillShowIndicator = true;
-		} }
-		scrollEnabled={ listProps.scrollEnabled }
-		onScroll={ ( event ) => {
-			this.latestContentOffsetY = event.nativeEvent.contentOffset.y;
-		} }
-	>
-		<List { ...listProps } />
-	</KeyboardAwareScrollView>
-);
+				}, 50 );
+			} }
+			onKeyboardWillShow={ () => {
+				keyboardWillShowIndicator.current = true;
+			} }
+			scrollEnabled={ scrollEnabled }
+			onScroll={ ( event ) => {
+				latestContentOffsetY.current =
+					event.nativeEvent.contentOffset.y;
+			} }
+			{ ...listProps }
```

Comment:
I wonder if this relocated prop spread is unintentionally overriding some of the props set above, or some of the locally set props are irrelevant. There appears to be only one usage of this component and it [sets several style properties itself](https://github.com/WordPress/gutenberg/blob/4aab1983badaf5b5e8607a84e4e10915de647458/packages/block-editor/src/components/block-list/index.native.js#L257-L276). We may need to merge the props or remove the locally set props. 

* `keyboardShouldPersistTaps`
* `style`
* `scrollViewStyle`
* `contentContainerStyle`

Reply:
Yep, good point, we should review what was the purpose of the different style ones (`style`, `scrollViewStyle` and `contentContainerStyle`) in the original component and decide in each case the best way to proceed, probably some of the local ones are no longer necessary.